client/state/themes/selectors.js

```diff
@@ -795,3 +802,55 @@ export function getRecommendedThemes( state ) {
 export function areRecommendedThemesLoading( state ) {
 	return state.themes.recommendedThemes.isLoading;
 }
+
+/**
+ * Checks if a theme has auto loading homepage feature.
+ *
+ * @param {object} state   Global state tree
+ * @param {string} themeId An identifier for the theme
+ * @returns {boolean} True if the theme has auto loading homepage. Otherwise, False.
+ */
+export function hasAutoLoadingHomepageFeature( state, themeId ) {
+	return includes(
+		getThemeTaxonomySlugs( getTheme( state, 'wpcom', themeId ), 'theme_feature' ),
+		FEATURE_AUTO_LOADING_HOMEPAGE
+	);
+}
+
+/**
+ * Return the theme ID which is going to be installed.
+ *
+ * @param {object} state   Global state tree
+ * @returns {string} Theme ID,
+ */
+export function getPreInstallingThemeId( state ) {
```

Comment:
We can't use the existing `isInstallingTheme()` selector?

Reply:
No, it's just before to install the theme. It means that the installing process didn't start yet, and it can be canceled. From the user experience perspective, it means that the user can cancel the theme installation.
Maybe we should rename the method, though.

Reply:
We should add this context in a comment within the code.